salesforce/utils.py

```diff
@@ -131,6 +131,24 @@ def get_soap_update_body(sobject, data):
     return update_body
 
 
+def get_soap_upsert_boxy(sobject, data, external_id):
```

Comment:
TODO(ant): this isn't robust enough to support upserting references to other objects by external ID, which we are going to need